src/components/Modals/NewCourse/NewCourse.vue

```diff
@@ -153,78 +115,55 @@ export default Vue.extend({
       this.$emit('updateSemProps', season, year);
     },
     getReqsRelatedToCourse(selectedCourse: MatchingCourseSearchResult) {
-      const relatedReqs = [];
-      const potReqs = [];
+      const relatedRequirements: RequirementWithID[] = [];
+      const potentialRequirements: RequirementWithID[] = [];
 
       // parse through reqs object
       for (let i = 0; i < this.reqs.length; i += 1) {
         const subreqs = this.reqs[i].ongoing;
         for (let j = 0; j < subreqs.length; j += 1) {
           // requirements
+
           const subRequirement = subreqs[j].requirement;
           if (subRequirement.fulfilledBy === 'courses') {
             const { courses } = subRequirement;
             for (let k = 0; k < courses.length; k += 1) {
               for (const [, ids] of Object.entries(courses[k])) {
                 if (ids.includes(selectedCourse.id)) {
-                  relatedReqs.push(subreqs[j].requirement.name);
+                  relatedRequirements.push({ id: subreqs[j].id, name: subRequirement.name });
                   break;
                 }
               }
             }
           }
           // potential requirements
           if (subreqs[j].fulfilledBy === 'self-check') {
-            potReqs.push(subreqs[j].requirement.name);
+            potentialRequirements.push({ id: subreqs[j].id, name: subRequirement.name });
           }
-          this.requirements = [...relatedReqs];
-          this.potentialReqs = [...potReqs];
-          this.selectedReqs = [...relatedReqs];
+          this.relatedRequirements = relatedRequirements;
+          this.potentialRequirements = potentialRequirements;
+          this.selectedRequirementIDs = relatedRequirements.map(it => it.id);
         }
       }
     },
-    editReq({ name, isSelected }: Readonly<{ name: string; isSelected: boolean }>) {
-      if (isSelected) {
-        this.selectedReqs.push(name); // add to selectedReqs
-      } else {
-        // remove from selectedReqs
-        const index = this.selectedReqs.indexOf(name);
-        if (index > -1) {
-          this.selectedReqs.splice(this.selectedReqs.indexOf(name), 1);
-        }
-      }
+    onSelectedChange(selected: readonly string[]) {
+      this.selectedRequirementIDs = selected;
     },
     next() {
       this.editMode = false;
-
-      // update potentialReqs by removing the ones that were selected
-      const newPotReqs = [];
-      for (let i = 0; i < this.potentialReqs.length; i += 1) {
-        if (!this.selectedReqs.includes(this.potentialReqs[i])) {
-          newPotReqs.push(this.potentialReqs[i]);
-        }
-      }
-      // add the requirements that were deselected to potential requirements
-      for (let i = 0; i < this.requirements.length; i += 1) {
-        if (!this.selectedReqs.includes(this.requirements[i])) {
-          newPotReqs.push(this.requirements[i]);
-        }
-      }
-      this.requirements = [...this.selectedReqs];
-      this.potentialReqs = [...newPotReqs];
```

Comment:
this code moves the selected potential requirement into related requirement. I'm not sure why we need this, since it can be very confusing (requirement switching category). It also interferes with the refactoring so I just removed it.